pytorch/pytorch-native/src/main/native/ai_djl_pytorch_jni_PyTorchLibrary_inference.cc

```diff
@@ -46,6 +46,75 @@ JNIEXPORT jlong JNICALL Java_ai_djl_pytorch_jni_PyTorchLibrary_moduleLoad(
   API_END_RETURN()
 }
 
+JNIEXPORT jlong JNICALL Java_ai_djl_pytorch_jni_PyTorchLibrary_moduleLoadFromIS(
+   JNIEnv* env, jobject jthis, jobject jis, jintArray jarray) {
+    API_BEGIN()
+    jclass is_class = env->GetObjectClass(jis);
+    if (is_class == nullptr) {
+        env->ThrowNew(NULL_PTR_EXCEPTION_CLASS, "Java inputStream class is not found");
+        return -1;
+    }
+    jmethodID method_id = env->GetMethodID(is_class, "read", "([BII)I");
+    if (method_id == nullptr) {
+        env->ThrowNew(ENGINE_EXCEPTION_CLASS, "The read method in InputStream is not found");
+        return -1;
+    }
+    std::ostringstream os;
+    int len = 4194304; // 4MB
+    jbyteArray arr = env->NewByteArray(len);
+    char* temp = new char[len];
+    int available = 0;
+    while(available != -1) {
+        available = env->CallIntMethod(jis, method_id, arr, 0, len);
+        if (available != -1) {
+            env->GetByteArrayRegion(arr, 0, available, reinterpret_cast<jbyte*>(temp));
+            os.write(temp, available);
+        }
+    }
+    std::istringstream in(os.str());
+    const torch::Device device = utils::GetDeviceFromJDevice(env, jarray);
+    const torch::jit::script::Module module = torch::jit::load(in, device);
+    const auto* module_ptr = new torch::jit::script::Module(module);
+    env->DeleteLocalRef(arr);
+    return reinterpret_cast<uintptr_t>(module_ptr);
+    API_END_RETURN()
+}
+
+JNIEXPORT void JNICALL Java_ai_djl_pytorch_jni_PyTorchLibrary_moduleWriteToOS(
+        JNIEnv* env, jobject jthis, jlong module_handle, jobject jos) {
+    API_BEGIN()
+    auto* module_ptr = reinterpret_cast<torch::jit::script::Module*>(module_handle);
+    std::ostringstream stream;
+    module_ptr->save(stream);
+    auto str = stream.str();
+    jclass os_class = env->GetObjectClass(jos);
```

Comment:
I think we can move class and method lookup into ai_djl_pytorch_jni_cache.h file